src/rpc/state_api.rs

```diff
@@ -196,9 +196,9 @@ pub(in crate::rpc) async fn state_miner_active_sectors<DB: Blockstore>(
 
     // Collect active sectors from each partition in each deadline.
     let mut active_sectors = vec![];
-    miner_state.for_each_deadline(policy, bs, |_dlidx, dl| {
-        dl.for_each(bs, |_partidx, part| {
-            active_sectors.push(part.active_sectors());
+    miner_state.for_each_deadline(policy, bs, |_dlidx, deadline| {
```

Comment:
We are basically repeating ourselves here, `Filecoin.StateMinerFaults` and perhaps another couple of RPC methods. In Lotus they are using callbacks to deal with that, not sure if we want to replicate this.

We could do something like:

`fn callback(enum: SectorType) -> FnMut.. {
   match SectorType {
      Active => sectors.push(partition.active_sectors());
...
    }
}` 

It's not a whole lot of code, but having a helper for this could be more comprehensive and less repetitive. I'm happy to take this on as a follow-up if it makes sense.

Reply:
Yeah, let's look at refactoring this soon.